### PR TITLE
Add clang-cuda support

### PR DESCRIPTION
**Pull request by davidbeckingsale**
Add support for clang's native CUDA compilation.

### CODE REVIEW

File: cmake/BLTOptions.cmake [hunk 1]
```diff
@@ -60,6 +60,9 @@ option(ENABLE_SHARED_LIBS  "Enables shared libraries." OFF)
 option(ENABLE_MPI          "Enables MPI support" OFF)
 option(ENABLE_OPENMP       "Enables OpenMP compiler support" OFF)
 option(ENABLE_CUDA         "Enable CUDA support" OFF)
+option(ENABLE_CLANG_CUDA   "Enable Clang's native CUDA support" OFF)
+mark_as_advanced(ENABLE_CLANG_CUDA)
+set(BLT_CUDA_ARCH "sm_30" CACHE STRING "Compute architecture to use when generating CUDA code")
```
Comment (davidbeckingsale) [inline cmake/BLTOptions.cmake]: I'm not sure the best way to handle this argument in normal CUDA builds - do we want to force it into `CUDA_NVCC_FLAGS`?

**Comment by davidbeckingsale** [comment]
@cyrush @white238 any thoughts on this?

File: cmake/BLTOptions.cmake (hunk 1 above)
Comment (cyrush) [inline cmake/BLTOptions.cmake]: To be consistent, we should use this for both cases (cuda-clang + nvcc), adding to NVCC FLAGS seems ok.

Or maybe make it optional, and have a default for cuda-clang?

**Comment by davidbeckingsale** [comment]
@cyrush - I updated this to make the "arch" argument specific to clang-cuda. I think this is the best approach for now.

**Pull request merged**